app/presenters/signup_url_hidden_params_presenter.rb

```diff
@@ -0,0 +1,39 @@
+class SignupUrlHiddenParamsPresenter
+  def initialize(content_item, view_context)
+    @content_item = content_item
+    @view_context = view_context
+  end
+
+  def hidden_params
+    @hidden_params ||= clean_filtered_params
+  end
+
+private
+
+  attr_reader :content_item, :view_context
+
+  def clean_filtered_params
+    @clean_filtered_params ||= begin
+      ParamsCleaner
+        .new(filtered_params)
+        .cleaned
+        .delete_if { |_, value| value.blank? }
+    end
+  end
+
+  def filtered_params
+    facet_choices = content_item['details'].fetch('email_filter_facets', []).reject { |facet| facet.key?('facet_choices') }
```

Comment:
Could we maybe extract a couple of these variables to private methods? It would help reduce the line length and make the `filtered_params` method more focused.